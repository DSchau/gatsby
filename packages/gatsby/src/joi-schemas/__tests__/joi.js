const { gatsbyConfigSchema } = require(`../joi`)

describe(`gatsby config`, () => {
  it(`strips trailing slashes from url fields`, async () => {
    const config = {
      pathPrefix: `/blog///`,
      assetPrefix: `https://cdn.example.com/`,
    }

    expect(await gatsbyConfigSchema.validate(config)).toEqual({
      pathPrefix: `/blog`,
      assetPrefix: `https://cdn.example.com`,
    })
  })

  it(`allows assetPrefix to be full URL`, async () => {
    const config = {
      assetPrefix: `https://cdn.example.com`,
    }

    expect(await gatsbyConfigSchema.validate(config)).toEqual(config)
  })

  it(`allows relative paths for url fields`, async () => {
    const config = {
      pathPrefix: `/blog`,
      assetPrefix: `/assets`,
    }

    expect(await gatsbyConfigSchema.validate(config)).toEqual(config)
  })

  it(`does not allow pathPrefix to be full URL`, async () => {
    const config = {
      pathPrefix: `https://google.com`,
    }

    try {
      await gatsbyConfigSchema.validate(config)
    } catch (e) {
      expect(e.message).toContain(`must be a valid relative uri`)
    } finally {
      expect.assertions(1)
    }
  })
})
