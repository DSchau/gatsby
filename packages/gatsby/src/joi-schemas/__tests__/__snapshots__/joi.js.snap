// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby config does not allow pathPrefix to be full URL 1`] = `"child \\"pathPrefix\\" fails because [\\"pathPrefix\\" must be a valid relative uri]"`;
